rfcs/accepted/0002-cd-update-snap.md

```diff
@@ -0,0 +1,89 @@
+- Feature Name: Update Jest Snap during release process
+- Start Date: 2023-04-25
+
+# Summary
+
+[summary]: #summary
+
+Move and reworks some release steps to update Jest Snap when we do a release of the CLI while not doubling the release time.
+
+# Motivation
+
+[motivation]: #motivation
+
+When a release of the CLI happens, it will now makes some Jest Snap outdated, which causes the tests to fail.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+Our current release process looks like this
+
+```mermaid
+graph LR
+    A(npm) --> B(Git) --> C(GitHub) --> D(Build)
+```
+
+1. We release to NPM
+2. We create and push a release commit and associated git tags
+3. We create a GitHub release for the CLI (if the CLI has changed)
+4. We build the CLI and add the binaries to the GitHub release
+
+We need the release commit to include the snap udpates to keep the main branch green.
+To do so while not hampering our release process too much, our release process would change to look like this:
+
+```mermaid
+graph LR
+    A(npm)  --> B(E2E)      --> C(Git) --> D(GitHub)
+    A       --> B2(Build)   --> D
+```
+
+1. We release to NPM
+2. The release process split into two branches:
+   - One branch start building the binaries right away.
+     - It stores them in a GitHub Job Artifact.
+     - It 'noop' if the CLI does not to be build.
+   - The other branch refresh the Jest Snap and then proceed with the previous release process up to the Git commit/push.
+3. The creation of the GitHub release now occurs when both Git and the binary jobs are done, and only if the CLI needs to be built/released.
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+- Split the Release Workflow into several workflows, one by step
+- NPM job would export a git patch so that E2E and Build can continue without any hitches.
+- E2E and Git can be one job split into two steps
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- Potential increase in complexity
+- Lack of control on our assertions: a bot will automatically update some assertions
```

Comment:
what are the "assertions" referring to here?

Reply:
```suggestion
- Lack of control over our assertions: a bot will automatically update the Jest Snaps, which are what we 'expect' the code to do. (e.g `expect(something).toMatchSnapshot()`)
```